packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -160,6 +110,39 @@ class StripeHelper {
     }
   }
 
+  /**
+   * Get a cached result at a cache key and regenerated it with `refreshFunction`
+   * if its expired.
+   *
+   * @param {string} cacheKey
+   * @param {() => Promise<T>} refreshFunction
+   * @returns {Promise<T | undefined>} possibly cached result
+   */
+  async getCachedResult(cacheKey, refreshFunction) {
+    try {
+      const json = await this.redis.get(cacheKey);
+      if (json) {
+        return JSON.parse(json);
+      }
+    } catch (err) {
+      this.log.error(`stripeHelper.getCachedResult.redis.get.failed`, { err });
+    }
+
+    const result = await refreshFunction();
+    try {
+      await this.redis.set(
+        cacheKey,
+        JSON.stringify(result),
+        'EX',
+        this.cacheTtlSeconds
```

Comment:
Do we *never* want to have different cache times for different data? I guess we could add the parameter back in if/when we end up doing that again

Reply:
As it stands, there is only one ttl defined, and I think that it's best to reduce the overhead of passing additional parameters when unnecessary. I actually would like to restructure how we are interacting with Stripe objects and caching entirely. I am working on a proposal for that.